### PR TITLE
Variables in Tags

### PR DESCRIPTION
**Pull request by NyCodeGHG**
Functionality for "variables" in tags like:

```
Hello this is my cool tag
$(http https://example.com/dings)
$(calc 1 + 5)
$(username)
```

### CODE REVIEW

**Comment by NyCodeGHG** [comment]
forgot floating numbers

**Review by DRSchlaubi — APPROVED** [review]
The two files i was able to review look fine

**Review by warriorzz — CHANGES_REQUESTED** [review]
Checks failed, but code is fine :)

**Comment by DRSchlaubi** [comment]
Looks like we need to upgrade the CI's server version

File: src/main/kotlin/de/nycode/bankobot/variables/parsers/calc/CalcExpression.kt [hunk 1]
```diff
@@ -37,24 +37,32 @@ class CalcExpression(val input: String) : Expression<BigDecimal> {
...
+            if (result != null) {
+                return result as BigDecimal
+            }
 
-        val input = CharStreams.fromString(input)
-        val lexer = CalculationLexer(input).apply {
-            removeErrorListeners()
-            addErrorListener(ThrowingErrorListener)
-        }
+            val input = CharStreams.fromString(input)
```
Comment (DRSchlaubi) [inline src/main/kotlin/de/nycode/bankobot/variables/parsers/calc/CalcExpression.kt:45]: Is there a reason for this ident?

**Review by warriorzz — APPROVED** [review]

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]

File: src/main/kotlin/de/nycode/bankobot/variables/parsers/calc/CalcExpression.kt (hunk 1 above)
Comment (NyCodeGHG) [reply src/main/kotlin/de/nycode/bankobot/variables/parsers/calc/CalcExpression.kt:45]: wtf it's not so in my ide
image

**Review by DRSchlaubi — APPROVED** [review]

**Pull request merged**